server/mediaserver.go

```diff
@@ -386,6 +404,25 @@ func jsonProfileToVideoProfile(resp *authWebhookResponse) ([]ffmpeg.VideoProfile
 	return profiles, nil
 }
 
+func jsonDetectionToDetectorProfile(resp *authWebhookResponse) core.DetectionConfig {
+	sceneClassification := ffmpeg.SceneClassificationProfile{
+		SampleRate: resp.Detection.SceneClassificationProfile.SampleRate,
+		Classes:    []ffmpeg.DetectorClass{},
+	}
+	for _, class := range resp.Detection.SceneClassificationProfile.Classes {
+		sceneClassification.Classes = append(sceneClassification.Classes,
+			ffmpeg.DetectorClass{
+				ID:   class.ID,
+				Name: class.Name,
+			})
+	}
+	detection := core.DetectionConfig{
+		Freq:     resp.Detection.Freq,
+		Profiles: []ffmpeg.DetectorProfile{&sceneClassification},
```

Comment:
Is a pointer to `sceneClassification` required when creating the `[]ffmeg.DetectorProfile` slice or would the struct value be fine as well?

Reply:
Currently the `DetectorProfile` interface is implemented for the pointer to the struct `*SceneClassificationProfile` https://github.com/livepeer/lpms/pull/228/files#diff-f3da988682c76502f83c01535ea7a53ae69403b27392275ef770e2bc69ce724aR28

In retrospect we should've implemented the interface directly for the struct type, but changing it now means touching 3 different PRs - are there any obvious cons to using a struct pointer here?
If yes, I don't mind changing the 3 PRs - If no, let's keep this as is for now.

Reply:
Implementing the interface with the pointer seems okay for now. Just wanted to be wary about mixing usage of the struct value and the struct pointer.